nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/AbstractAzureBlobProcessor_v12.java

```diff
@@ -0,0 +1,163 @@
+import static org.apache.nifi.processors.azure.storage.utils.BlobAttributes.ATTR_NAME_BLOBTYPE;
+import static org.apache.nifi.processors.azure.storage.utils.BlobAttributes.ATTR_NAME_CONTAINER;
+import static org.apache.nifi.processors.azure.storage.utils.BlobAttributes.ATTR_NAME_ETAG;
+import static org.apache.nifi.processors.azure.storage.utils.BlobAttributes.ATTR_NAME_LANG;
+import static org.apache.nifi.processors.azure.storage.utils.BlobAttributes.ATTR_NAME_LENGTH;
+import static org.apache.nifi.processors.azure.storage.utils.BlobAttributes.ATTR_NAME_MIME_TYPE;
+import static org.apache.nifi.processors.azure.storage.utils.BlobAttributes.ATTR_NAME_PRIMARY_URI;
+import static org.apache.nifi.processors.azure.storage.utils.BlobAttributes.ATTR_NAME_TIMESTAMP;
+
+public abstract class AbstractAzureBlobProcessor_v12 extends AbstractProcessor {
+
+    public static final PropertyDescriptor STORAGE_CREDENTIALS_SERVICE = new PropertyDescriptor.Builder()
+            .name("storage-credentials-service")
+            .displayName("Storage Credentials")
+            .description("Controller Service used to obtain Azure Blob Storage Credentials.")
+            .identifiesControllerService(AzureStorageCredentialsService_v12.class)
+            .required(true)
+            .build();
+
+    public static final PropertyDescriptor BLOB_NAME = new PropertyDescriptor.Builder()
+            .name("blob-name")
+            .displayName("Blob Name")
+            .description("The full name of the blob")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .required(true)
+            .build();
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("All successfully processed FlowFiles are routed to this relationship")
+            .build();
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("Unsuccessful operations will be transferred to the failure relationship.")
+            .build();
+
+    private static final Set<Relationship> RELATIONSHIPS = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
+            REL_SUCCESS,
+            REL_FAILURE
+    )));
+
+    private BlobServiceClient storageClient;
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @OnScheduled
+    public void onScheduled(ProcessContext context) {
+        storageClient = createStorageClient(context);
+    }
+
+    @OnStopped
+    public void onStopped() {
+        storageClient = null;
+    }
+
+    protected BlobServiceClient getStorageClient() {
+        return storageClient;
+    }
+
+    public static BlobServiceClient createStorageClient(PropertyContext context) {
```

Comment:
I'm afraid we could not hide it totally. Even if it was `protected`, someone could use this abstract class as a parent class for their custom processors. Unfortunately, we will not be able to change it due to backward compatibility.

Furthermore, it is `public` and `static` because of the Listen processor. Put, Fetch and Delete processors are subclasses of this abstract class (so `protected` would be fine) but Listen should come from `AbstractListProcessor`. For this reason, it accesses this method in static way. I know it is weird but the existing blob/adls List processors use the same and I have no better idea. I tried to subclass the Listen processor from the blob abstract class and define an `AbstractListProcessor` sublcass in the Listen processor (as an inner class) and use it as a delegate but it is not straightforward because all the processor lifecycle method callbacks should be delegated as well. So I gave up this approach.
Any idea would be appreciated!